src/Promitor.Integrations.Sinks.Prometheus/PrometheusScrapingEndpointMetricSink.cs

```diff
@@ -94,7 +94,10 @@ private IMetricFamily<IGauge> CreateGauge(string metricName, string metricDescri
 
             if (measuredMetric.IsDimensional)
             {
-                labels.Add(measuredMetric.DimensionName.SanitizeForPrometheusLabelKey(), measuredMetric.DimensionValue);
+                for (var i = 0; i < measuredMetric.DimensionNames.Count; i++)
```

Comment:
We did not like the for-loop much either, but the problem with a foreach was that we also need to iterate through both `measuredMetric.DimensionNames` and `measuredMetric.DimensionValues` at the same time and using `zip` made the code look more obtuse, so we opted for the classic for-loop.
However, with the new `MeasuredMetricDimension`, this problem is solved, since we can now just iterate through `measuredMetric.Dimensions` and get both the name and value.